pytorch_lightning/strategies/ipu.py

```diff
@@ -260,6 +266,18 @@ def to_tensor(x):
         args = apply_to_collection(args, dtype=(int, float), function=to_tensor)
         return args
 
+    def _disable_zero_grad(self) -> None:
+        if is_overridden("optimizer_zero_grad", self.lightning_module):
+            warning_cache.warn(
+                "You have overridden the `LightningModule.optimizer_zero_grad` hook but it will be ignored since"
+                " IPUs handle the zeroing of gradients internally."
+            )
+
+        def override_fn(*args: Any, **kwargs: Any) -> None:
+            pass
+
+        self.lightning_module.optimizer_zero_grad = override_fn  # type: ignore[assignment]
```

Comment:
Same comment as https://github.com/PyTorchLightning/pytorch-lightning/pull/12842#discussion_r860843496

Reply:
Pushed the change in https://github.com/PyTorchLightning/pytorch-lightning/pull/12913/commits/80f092070d45742a024ac893539ac09a699c6958